app/javascript/components/clients/Doctorappointment.js

```diff
@@ -0,0 +1,71 @@
+/* eslint-disable no-restricted-globals */
+/* eslint-disable no-alert */
+/* eslint-disable jsx-a11y/label-has-associated-control */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+import React from 'react';
+import { useParams } from 'react-router-dom';
+import { useSelector } from 'react-redux';
+
+import './Bookappointment.css';
+
+const DoctorAppointment = () => {
+  const { id } = useParams();
+  const appointments = useSelector((state) => state.appointmentReducer);
+
+  const onSubmit = (e) => {
+    document.querySelector('.popup-box').style.display = 'none';
+    const date = document.getElementById('appointment_date').value;
+    const time = document.getElementById('appointment_time').value;
+    const doctorId = parseInt(id, 10);
+    if ((appointments.filter((appointment) => appointment.appointment_time === time).length > 0)
```

Comment:
Nice way to prevent doubled appointments!

Reply:
Thanks!